Source/Scene/Cesium3DTileStyleEngine.js

```diff
@@ -0,0 +1,85 @@
+/*global define*/
+define([
+        '../Core/defined',
+        '../Core/defineProperties'
+    ], function(
+        defined,
+        defineProperties) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    function Cesium3DTileStyleEngine() {
+        this._style = undefined;      // The style provided by the user
+        this._styleDirty = false;     // true when the style is reassigned
+        this._lastStyleTime = 0;      // The "time" when the last style was assigned
+    }
+
+    defineProperties(Cesium3DTileStyleEngine.prototype, {
+        style : {
+            get : function() {
+                return this._style;
+            },
+            set : function(value) {
+                this._style = value;
+                this._styleDirty = true;
+            }
+        }
+    });
+
+    Cesium3DTileStyleEngine.prototype.makeDirty = function() {
+        this._styleDirty = true;
+    };
+
+    Cesium3DTileStyleEngine.prototype.applyStyle = function(tileset, frameState) {
+        if (!tileset.ready) {
+            return;
+        }
+
+        if (defined(this._style) && !this._style.ready) {
+            return;
+        }
+
+        var styleDirty = this._styleDirty;
+
+        if (frameState.passes.render) {
+            // Don't reset until the color pass, e.g., for mouse-over picking
+            this._styleDirty = false;
+        }
+
+        if (styleDirty) {
+            // Increase "time", so the style is applied to all visible tiles
```

Comment:
Since I can't think of something better, I think I'm going to stick with this. If it helps justify my case at least one other area in Cesium uses this convention, `DrawCommand` has a `lastDirtyTime`.